### PR TITLE
fix(`UPL-63`): scream if node < 18

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes UPL-63

### CODE REVIEW

File: packages/uploadthing/src/internal/incompat-node-guard.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        .substring(1)
+        .split(".")
+        .map((v) => parseInt(v, 10));
+    }
+  }
+
+  if (major && major < 18) {
+    throw new Error(
+      `[UT]: YOU ARE USING A LEGACY (${major}) NODE VERSION WHICH ISN'T OFFICIALLY SUPPORTED. PLEASE UPGRADE TO NODE 18+.`,
+    );
```
Comment (markflorkowski) [inline packages/uploadthing/src/internal/incompat-node-guard.ts:25]:
```suggestion
    );
    process.exit(1);  // Kill the process if it isn't going to work correctly anyway
```

**Review by markflorkowski — APPROVED** [review]

File: packages/uploadthing/src/internal/incompat-node-guard.ts (hunk 1 above)
Comment (juliusmarminge) [reply packages/uploadthing/src/internal/incompat-node-guard.ts:25]: Is that cross-runtime compatible?

Comment (markflorkowski) [reply packages/uploadthing/src/internal/incompat-node-guard.ts:25]: Good point. I would assume so, but don't actually know for sure. I'll have to check.

Comment (markflorkowski) [reply packages/uploadthing/src/internal/incompat-node-guard.ts:25]: Should work in bun. Not sure of other runtimes that have `process` defined tbh

Comment (juliusmarminge) [reply packages/uploadthing/src/internal/incompat-node-guard.ts:25]: We can fix that when we get there if it's an issue I guess

**Pull request merged**